clients/java/signalr/src/main/java/com/microsoft/aspnet/signalr/HubConnection.java

```diff
@@ -151,6 +153,45 @@ public HubConnectionState getConnectionState() {
         return connectionState;
     }
 
+
+    public CompletableFuture startAsync() throws Exception {
+        if (connectionState != HubConnectionState.DISCONNECTED) {
+            return CompletableFuture.completedFuture(null);
+        }
+        if (!skipNegotiate) {
+            int negotiateAttempts = 0;
+            do {
+                accessToken = (negotiateResponse == null) ? null : negotiateResponse.getAccessToken();
+                negotiateResponse = handleNegotiate();
```

Comment:
Make sure there's a bug tracking async negotiation unless you're going to do it in this PR.